### ISSUE TITLE
Which python version to use?

### ISSUE THREAD

**Issue by siddharth108s** [0ssamaak0/CLIPPyX#17]
I tried using Python 3.12 and 3.10 using virtual environment but was unable to run the app. It says unable to import Flask even though all dependencies are installed.

**Comment by 0ssamaak0** [comment]
Can you share the error?
Note that the current installation steps are kinda outdated. I will update it soon with the next release

**Comment by siddharth108s** [comment]
(venv) PS E:\CLIPPyX> CLIPPyX
Traceback (most recent call last):
  File "E:\CLIPPyX\create_index.py", line 1, in <module>
    from Index.create_db import *
  File "E:\CLIPPyX\Index\create_db.py", line 1, in <module>
    import chromadb
ModuleNotFoundError: No module named 'chromadb'
Traceback (most recent call last):
  File "E:\CLIPPyX\server.py", line 1, in <module>
    from flask import Flask, abort, request, jsonify, send_from_directory
ModuleNotFoundError: No module named 'flask'
Traceback (most recent call last):
  File "\\?\E:\CLIPPyX\venv\Scripts\CLIPPyX-script.py", line 33, in <module>
    sys.exit(load_entry_point('CLIPPyX', 'console_scripts', 'CLIPPyX')())
TypeError: 'module' object is not callable

**Comment by siddharth108s** [comment]
Complete PowerShell input and output here: https://[pastebin.com/raw/rmkninfL](https://pastebin.com/raw/rmkninfL)

**Comment by 0ssamaak0** [comment]
There's a problem with your python environment. Can you show me the output of `pip list`? Or try to create a clean installation in clean env (preferably conda)

**Comment by siddharth108s** [comment]
> There's a problem with your python environment. Can you show me the output of `pip list`? Or try to create a clean installation in clean env (preferably conda)

https://pastebin.com/CSmFmJFf

**Comment by 0ssamaak0** [comment]
I think I got it. You have many Python versions installed. In `main.py`, it runs these commands:
```python
subprocess.run(["python", "create_index.py"])
subprocess.run(["python", "server.py"])
```

which run in the default Python you have rather than this venv.

Please verify if this is the scenario so I can fix it.

**Comment by siddharth108s** [comment]
> I think I got it. You have many Python versions installed. In `main.py`, it runs these commands:
> 
> ```python
> subprocess.run(["python", "create_index.py"])
> subprocess.run(["python", "server.py"])
> ```
> 
> which run in the default Python you have rather than this venv.
> 
> Please verify if this is the scenario so I can fix it.

Yes! This was the issue it seems.
I replace them with:
`subprocess.run([python_executable, "create_index.py"])
subprocess.run([python_executable, "server.py"])`

and now it works

**Comment by 0ssamaak0** [comment]
Thanks for pointing to this issue, I will try to modify the main to avoid this problem! I will keep the issue open until I fix it

**Comment by 0ssamaak0** [comment]
Fixed in 2124eddc541dee15cb694a9afcd162a3eb62ab6f
